### PR TITLE
Existing PVC support for stable/drupal

### PR DESCRIPTION
**Pull request by scottrigby**
## Context

@prydonius This is what we talked about in relation to #1057 (this is an independent PR, but which the other would rely on if we go the PV/PVC route for local k8s app dev best practices) - just adding existing PVC support to an example stable chart (in this case Drupal. If this works out we can open PRs for the other applicable stable charts missing this feature).

## Testing
I have not yet tested this locally, because I'm having trouble getting a PVC to bind to a PV in minikube for some reason. If it's helpful, Here are YAML files for the PV and PVC I'm attempting to bind in Minikube.

### existing-pv.yaml
<details>

```yaml
kind: PersistentVolume
apiVersion: v1
metadata:
  name: existing-drupal-pv
  labels:
    environment: local
  annotations:
    volume.alpha.kubernetes.io/storage-class: default
spec:
  capacity:
    storage: "8Gi"
  accessModes:
    - ReadWriteOnce
  persistentVolumeReclaimPolicy: Recycle
  hostPath:
    path: "/PATH/TO/EXISTING/DIR/ON/HOST/MACHINE"
```
</details>

### existing-pvc.yaml
<details>

```yaml
kind: PersistentVolumeClaim
apiVersion: v1
metadata:
  name: existing-drupal-pvc
  labels:
    environment: local
  annotations:
    volume.alpha.kubernetes.io/storage-class: default
spec:
  accessModes:
    - ReadWriteOnce
  resources:
    requests:
      storage: "8Gi"
  selector:
    matchLabels:
      environment: local
```
</details>

I'm hoping someone more knowledgable about PV/PVC binding may be able to help with this part. If so, I'll be happy to add testing feedback. (note that I have read the [PersistentVolume Selector section](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#selector) and [Labels and Selectors docs](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/), and have found very few other discussions about this online, mostly old ones like [this](http://stackoverflow.com/questions/34282704/can-a-pvc-be-bound-to-a-specific-pv/34323691#34323691)).

### CODE REVIEW

**Comment by scottrigby** [comment]
OK, I had fixed the pvc name default typo. And I was able to test this. This is still ready for review by a contributor, but it tests out 👍 CC @prydonius

**Comment by scottrigby** [comment]
Here is a working example PV/PVC, so other people can test on minikube:

## Test steps
1. Enable minikube as usual (I use xhyve, but AFAIK it should work with other drivers)
1. `kubectl create -f existing-drupal-pv.yaml` with this PV file:
    ```yaml
    kind: PersistentVolume
    apiVersion: v1
    metadata:
      name: existing-drupal-pvc
    spec:
      capacity:
        storage: 8Gi
      accessModes:
        - ReadWriteOnce
      hostPath:
        path: "/FULL/PATH/TO/EXISTING/HOST/DIRECTORY"
    ```
    Change your `hostPath.path` to match your host machine's actual path. For example, the path I am testing on MacOS is `/Users/scottrigby/development/github.com/kubernetes/charts/existing-drupal-pvc`, but you can choose any path you want as long as the directory exists and is writable.
1. `kubectl create -f existing-drupal-pvc.yaml` with this PVC file:
    ```yaml
    kind: PersistentVolumeClaim
    apiVersion: v1
    metadata:
      name: existing-drupal-pvc
    spec:
      storageClassName: ""
      accessModes:
        - ReadWriteOnce
      resources:
        requests:
          storage: 8Gi
    ```
1. Check out this PR locally (with [hub](https://github.com/github/hub) you can do `hub checkout https://github.com/kubernetes/charts/pull/1084`
1. Install the chart with the new parameter `helm install ./stable/drupal/ --set persistence.ExistingClaim="existing-drupal-pvc"` (if you use the above example files. Or whatever your existing claim name is)

**Comment by lypht** [comment]
I was able to test this successfully, and wanted to add that if anyone is running into permissions issues with the app not displaying the css correctly, the following steps will allow for this chart to successfully render drupal using a host path:
1. `chmod -R a+rwx /path/to/host/directory`
1. Then exec into the container: `kubectl exec -it POD_NAME bash` and
1. Change to the drupal directory: `cd /opt/bitnami/drupal` then
1. Clear the drupal registry: `drush cr`

You should be able to reload the page and render correctly.

**Comment by scottrigby** [comment]
@lypht @prydonius I changed `persistence.ExistingClaim` to `persistence.drupal.ExistingClaim` because this only applies to the `drupal-data` volume. I realized in retrospect the initial name was value was too general in the `.Values` tree, and may have been misleading.

File: stable/drupal/values.yaml [hunk 1]
```diff
@@ -93,6 +93,12 @@ persistence:
     accessMode: ReadWriteOnce
     size: 8Gi
 
+    ## A manually managed Persistent Volume Claim
+    ## Requires persistence.enabled: true
+    ## If defined, PVC must be created manually before volume will be bound
+    ##
+    # ExistingClaim:
```
Comment (prydonius) [inline stable/drupal/values.yaml]: rename to `existingClaim` to match case format

**Review by prydonius — CHANGES_REQUESTED** [review]
sorry it took me a while to get to this. Only a small change for consistency needed - also can you bump the chart minor version?

thanks @lypht for reviewing btw :)

**Comment by scottrigby** [comment]
@prydonius Updated `existingClaim` case, and bumped minor version.

**Review by prydonius — APPROVED** [review]

**Comment by prydonius** [comment]
@k8s-bot ok to test

**Pull request merged**